test/Npgsql.Tests/PoolTests.cs

```diff
@@ -397,35 +349,42 @@ public void Open_physical_failure()
             Port = 44444,
             MaxPoolSize = 1
         }.ToString();
-        using var conn = CreateConnection(connString);
-        for (var i = 0; i < 1; i++)
-            Assert.That(() => conn.Open(), Throws.Exception
-                .TypeOf<NpgsqlException>()
-                .With.InnerException.TypeOf<SocketException>());
-        Assert.True(PoolManager.Pools.TryGetValue(connString, out var pool));
-        AssertPoolState(pool, open: 0, idle: 0);
+        using var conn = new NpgsqlConnection(connString);
```

Comment:
Can use data source for this test no?

Reply:
Yeah, missed that one (there're probably still quite a few tests where we were creating the connection via a raw `NpgsqlConnection` instead of `CreateConnection` and `OpenConnection`.